File: src/test/kotlin/org/elasticsearch/bootstrap/JarHell.java
package org.elasticsearch.bootstrap;

import org.elasticsearch.common.SuppressForbidden;

import java.io.IOException;
import java.net.URISyntaxException;
import java.net.URL;
import java.nio.file.Path;
import java.util.Collections;
import java.util.Map;
import java.util.Set;
import java.util.function.Consumer;
import java.util.jar.Manifest;

public class JarHell {

    private JarHell() {
    }

    public static void checkJarHell(Consumer<String> output) throws IOException, URISyntaxException {
    }

    public static Set<URL> parseClassPath() {
        return Collections.emptySet();
    }

    @SuppressForbidden(reason = "resolves against CWD because that is how classpaths work")
    static Set<URL> parseClassPath(String classPath) {
        return Collections.emptySet();
    }

    @SuppressForbidden(reason = "needs JarFile for speed, just reading entries")
    public static void checkJarHell(Set<URL> urls, Consumer<String> output) throws URISyntaxException, IOException {
    }

    /**
     * inspect manifest for sure incompatibilities
     */
    private static void checkManifest(Manifest manifest, Path jar) {
    }

    public static void checkVersionFormat(String targetVersion) {
    }

    /**
     * Checks that the java specification version {@code targetVersion}
     * required by {@code resource} is compatible with the current installation.
     */
    public static void checkJavaVersion(String resource, String targetVersion) {
    }

    private static void checkClass(Map<String, Path> clazzes, String clazz, Path jarpath) {
    }
}

File: src/test/kotlin/org/elasticsearch/bootstrap/JarHell.kt
package org.elasticsearch.bootstrap

import java.net.URL
import java.nio.file.Path
import java.util.*
import java.util.Map
import java.util.Set
import java.util.function.Consumer
import java.util.jar.Manifest


object JarHell {
    @Throws(Exception::class)
    fun checkJarHell() {
    }

    @Throws(Exception::class)
    fun checkJarHell(urls: Array<URL?>?) {
    }

    @Throws(Exception::class)
    fun checkJarHell(urls: Set<URL?>?, output: Consumer<String?>?) {
    }

    fun checkVersionFormat(targetVersion: String?) {}
    fun checkJavaVersion(resource: String?, targetVersion: String?) {}
    fun parseClassPath(): Set<URL> {
        val urlElements: Set<URL> = LinkedHashSet()
        return urlElements
    }

    private fun checkManifest(manifest: Manifest, jar: Path) {}
    private fun checkClass(clazzes: Map<String, Path>, clazz: String, jarpath: Path) {}
}

